torch/csrc/jit/fusion_compiler.cpp

```diff
@@ -566,6 +882,18 @@ void CompiledFusionFunction::launch_with_tensors(at::ArrayRef<at::Tensor> inputs
       }
     }
   }
+
+  #ifdef USE_CUDA
+    // If the kernel call contains a random op, we need to pass in random seeds as
+    // well.
+    if(has_random && this->backend() == at::kCUDA) {
+      auto gen_ = THCRandom_getGenerator(at::globalContext().getTHCState());
+      uint64_t offset = gen_->state.philox_seed_offset.fetch_add(20);
```

Comment:
Can you explain where this `20` comes from and why it is correct?

Reply:
Actually I wasn't quite sure what to put here as well. The reference I had is pytorch/aten/src/ATen/native/cuda/Distributions.cu. When I talked to other developers about this, they told me "10 or 20 is a rough estimate on how many times the generator will be called. It's not ideal, but authors of those functions decided that it is good enough.".

Reply:
Unlike the other uses, the number of times an inner loops will run in a fusion kernel is dependent on the size of the tensors and the number of calls to rand.

In that case shouldn't it be  (number of times a single thread can call the random number generator)/4?

Max times a thread runs its inner loop is:

```
max_times_an_inner_loop_runs = ceil(numel/(num_threads_per_block*number_of_blocks_launched))
max_rand_calls_per_thread = num_calls_to_rand_in_body*max_times_an_inner_loop_runs
```

Reply:
It shouldn't be divided by 4, I think. Philox counter the way it's implemented here is counting every number. See in my pr for offset increment https://github.com/pytorch/pytorch/pull/9666/files#diff-13a03e6a295e68ffd7ca94678b5b581dR108
Also, Distributions.cu have a very inefficient pattern of calling random number generation, efficient pattern is in dropout pr also.

Reply:
I believe that 10 or 20 estimate was specific to the case of generating Poisson-distributed random numbers.  

If you're just generating uniformly distributed floats or ints one at a time, you're manually invoking the generator, so you know exactly how many times you're calling the generator per thread, as @zdevito says.  

The original tensorflow Philox anticipates that `offset` is a multiple of 4, where `offset` is the number of 32-bit values generated so far (aka, in prior kernel launches) for this particular subsequence.  We should stick to that convention to be safe.  You're already dividing `offset` by 4 [when you construct PhiloxRandom](https://github.com/pytorch/pytorch/pull/9795/files#diff-c76c51fc543bd5257d3ae9bb400bdbd8R357) within PhiloxWrapper, which is fine.  Since `offset` here is